### ISSUE TITLE
Support for `vim.lsp`

### ISSUE THREAD

**Issue by noib3** [noib3/nvim-oxi#47]
https://github.com/noib3/nvim-oxi/discussions/46?converting=1#discussion-4263989

**Comment by willothy** [comment]
Seems like these would require bindings directly to lua by fetching globals, since a lot of the LSP/Diagnostics stuff is implemented in Lua. Would you be open to someone contributing that? That's the approach I was taking with `nvim-utils`, but honestly I'd rather contribute those features to `oxi` and archive `utils` as it's not getting much use and `oxi` provides a better (and faster) API.

**Comment by noib3** [comment]
> Seems like these would require bindings directly to lua by fetching globals, since a lot of the LSP/Diagnostics stuff is implemented in Lua.

Yes, that'd be the approach. [This](https://github.com/noib3/nvim-oxi/blob/main/crates/oxi-diagnostic/src/diagnostic.rs#L6-L32) is a *very* rough sketch of such a binding that I wrote for `vim.diagnostic.enable`.

> Would you be open to someone contributing that?

Yes, although I should mention that I don't really have the time to focus on oxi right now (and probably won't for a few months). What this means practically is that you might have to wait a few days to receive feedback on the relative PRs.

**Comment by willothy** [comment]
No worries! I don't actively need nvim LSP features in Rust so there's no rush, just think it would be nice to have and would be happy to contribute :)

**Comment by willothy** [comment]
Cool! Would be nice to have those implemented in Rust.

**Comment by poljar** [comment]
> I should also mention that I was planning to rewrite Neovim's `vim.lsp` + `nvim-lspconfig` + `lspsaga.nvim` + `any-lsp-related-stuff-that-should-be-builtin-instead-of-being-relegated-to-a-3rd-party-plugin` in Rust as part of a new project, hopefully before EOY.

Is there any more info about this effort now? Any repo we can follow or a dedicated issue?

**Comment by noib3** [comment]
> Is there any more info about this effort now?

Not at the moment. I still intend to write that code but I'm currently focusing on another project. I'll post here when I have an update.
